### PR TITLE
Add relationships for transformers.

### PR DESCRIPTION
**Pull request by cmfcruz**
Adds the following relationships:
- Any was transformed into any
- Contract repository has member any
- Task has result any

Change-type: patch
Signed-off-by: Carlo Miguel F. Cruz <carloc@balena.io>

### CODE REVIEW

**Comment by joshbwlng** [comment]
@cmfcruz Tests failing with:
```
Error: File lib/contracts/relationship-any-was-transformed-to-any.ts hasn't been formatted with prettier
```

Which means you haven't ran `npm run lint:fix`. This should run automatically on git commit, unless you didn't install deps with `npm install`.

**Comment by joshbwlng** [comment]
Also, we should bring the `contract-repository` type contract in from `plugin-product-os` with this PR as its adding a relationship that references it. There's no reason for any transformer-related type contracts to be defined in `plugin-product-os` anyway.

**Comment by joshbwlng** [comment]
I'm taking care of the above, but noticed there are a few more things to confirm before merging.

I found the following relationships in `plugin-product-os`, do we need to keep these?
- https://github.com/product-os/jellyfish-plugin-product-os/blob/joshbwlng/drop-transformer-contracts/lib/contracts/relationship-contract-repository-latest-any.ts
- https://github.com/product-os/jellyfish-plugin-product-os/blob/joshbwlng/drop-transformer-contracts/lib/contracts/relationship-loop-owns-contract-repository.ts

There are some other transformer related relationships here in the worker, can we drop these or do we need to keep them?
- https://github.com/product-os/jellyfish-worker/blob/master/lib/contracts/relationship-task-generated-any.ts
- https://github.com/product-os/jellyfish-worker/blob/master/lib/contracts/relationship-transformer-generated-task.ts
- https://github.com/product-os/jellyfish-worker/blob/master/lib/contracts/relationship-transformer-worker-owns-task.ts

**Comment by joshbwlng** [comment]
@balena-ci retest

File: lib/contracts/relationship-any-was-transformed-to-any.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import type { RelationshipContractDefinition } from 'autumndb';
+
+export const relationshipAnyWasTransformedToAny: RelationshipContractDefinition =
+	{
+		slug: 'relationship-any-was-trasnformed-to-any',
+		type: 'relationship@1.0.0',
+		name: 'was transformed to',
```
Comment (joshbwlng) [inline lib/contracts/relationship-any-was-transformed-to-any.ts:7]: Should this be `was transformed into`? I see other references to this relationship are `was transformed into`, which doesn't match up.

Comment (joshbwlng) [reply lib/contracts/relationship-any-was-transformed-to-any.ts:7]: Also the slug has a typo: `trasnformed` => `transformed`

**Comment by joshbwlng** [comment]
Confirmed that the unit tests are failing because the link constraints defined in `jellyfish-client-sdk` don't match up. You could add those in to get those to pass, but we're in the process of dropping those entirely so I'd just wait a couple of days until we wrap that work up.

**Comment by joshbwlng** [comment]
Dropped unnecessary dependency on `jellyfish-client-sdk` and its link constraints with https://github.com/product-os/jellyfish-worker/pull/1618. Which means this *should* be mergeable once the above issues are answered and dealt with.

File: lib/contracts/relationship-any-was-transformed-to-any.ts (hunk 1 above)
Comment (cmfcruz) [reply lib/contracts/relationship-any-was-transformed-to-any.ts:7]: @joshbwlng , good find.  I fixed the typo and set all references to the relationship to `was transformed to`.

**Comment by cmfcruz** [comment]
@balena-ci rebase

**Comment by cmfcruz** [comment]
@balena-ci rebase

**Comment by joshbwlng** [comment]
@cmfcruz Integration tests are failing, related to changes in this PR:
```
sut_1       |   ● transformers › should create a task if a transformer matches a contract that changed artifactReady:false->true
sut_1       | 
sut_1       |     The wait query did not resolve
sut_1       | 
sut_1       |       134 | 	): Promise<T> => {
sut_1       |       135 | 		if (times === 0) {
sut_1       |     > 136 | 			throw new Error('The wait query did not resolve');
sut_1       |           | 			      ^
sut_1       |       137 | 		}
sut_1       |       138 | 		const results = await autumndbTestContext.kernel.query<T>(
sut_1       |       139 | 			autumndbTestContext.logContext,
sut_1       | 
sut_1       |       at waitForMatch (lib/test-utils.ts:136:10)
sut_1       |       at waitForMatch (lib/test-utils.ts:149:10)
sut_1       |       at Object.<anonymous> (test/integration/transformers.spec.ts:140:3)
```

**Comment by joshbwlng** [comment]
Found this error as well:
```
2022-07-08T08:11:15.223Z [autumndb] [crit] [CORE-TEST-40b1bb29-e2e3-459d-a701-279302dac1bc]: Relationship not found triggered-action@1.0.0 owns (inverse is owned by) task@1.0.0 for contract link-0e752c42-b5b7-4c81-ba0a-503e672f5304
2022-07-08T08:11:15.275Z [worker] [warning] [CORE-TEST-40b1bb29-e2e3-459d-a701-279302dac1bc]: Execute error {"error":{"message":"Relationship not found triggered-action@1.0.0 owns (inverse is owned by) task@1.0.0 for contract link-0e752c42-b5b7-4c81-ba0a-503e672f5304","name":"JellyfishUnknownRelationship","expected":true,"stack":"JellyfishUnknownRelationship: Relationship not found triggered-action@1.0.0 owns (inverse is owned by) task@1.0.0 for contract link-0e752c42-b5b7-4c81-ba0a-503e672f5304\n    at Kernel.preUpsert (/home/josh/git/product-os/jellyfish-worker/node_modules/autumndb/lib/kernel.ts:992:22)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at Kernel.insertContract (/home/josh/git/product-os/jellyfish-worker/node_modules/autumndb/lib/kernel.ts:777:3)\n    at /home/josh/git/product-os/jellyfish-worker/lib/index.ts:677:20\n    at Worker.commit (/home/josh/git/product-os/jellyfish-worker/lib/index.ts:1492:28)\n    at handler (/home/josh/git/product-os/jellyfish-worker/lib/actions/action-matchmake-task.ts:120:2)\n    at Worker.execute (/home/josh/git/product-os/jellyfish-worker/lib/index.ts:1351:22)\n    at flush (/home/josh/git/product-os/jellyfish-worker/lib/test-utils.ts:125:18)\n    at Object.flushAll (/home/josh/git/product-os/jellyfish-worker/lib/test-utils.ts:155:5)\n    at Object.<anonymous> (/home/josh/git/product-os/jellyfish-worker/test/integration/transformers.spec.ts:109:3)"},"input":{"id":"cc9db38c-007a-4624-a3f1-bd640b4856ab"},"action":"action-matchmake-task","time":19}
2022-07-08T08:11:15.291Z [worker] [info] [CORE-TEST-40b1bb29-e2e3-459d-a701-279302dac1bc]: Execute event posted {"slug":"execute-d95fda1c-4638-4d87-bf2f-ee1a66a6840f","type":"execute@1.0.0","target":"d95fda1c-4638-4d87-bf2f-ee1a66a6840f","actor":"53917a42-c299-483c-a121-591792214717","payload":{"card":"cc9db38c-007a-4624-a3f1-bd640b4856ab","error":true}}
```

Looks like we need a `triggered-action owns task` relationship?

**Comment by joshbwlng** [comment]
@balena-ci rebase

**Comment by joshbwlng** [comment]
@balena-ci rebase

**Comment by cmfcruz** [comment]
@balena-ci I self-certify!

**Pull request merged**